packages/build/src/plugins_core/secrets_scanning/index.ts

```diff
@@ -73,7 +77,7 @@ const coreStep: CoreStepFunction = async function ({
 
   const keysToSearchFor = getSecretKeysToScanFor(envVars, passedSecretKeys)
 
-  if (keysToSearchFor.length === 0 && !enhancedSecretScan) {
+  if (keysToSearchFor.length === 0 && (!enhancedSecretScan || !enhancedScanShouldImpactBuilds)) {
```

Comment:
Just checking here - the change of condition here is to address this part?

> I think if we did this, we could only do it for projects which are already being scanned for secrets (because they have env vars marked as secret) because otherwise we'd be adding a new step to the build process for folks without providing any context on it.

If so - it shamefully took me a while to grasp how this should work 😅 , but if this is to support above I'd suggest equaivalent that _might_ be easier to parse:
```suggestion
  if (keysToSearchFor.length === 0 && !(enhancedSecretScan && enhancedScanShouldImpactBuilds)) {
```

But if it's more readable for you the way it is - feel free to keep it as is

Reply:
Yeah you're right that's more readable - I'll push a change!